aws_emr_cluster Panic with Missing StateChangeReason Code/Message

This is not consistently occurring (since it depends on a failing configuration), but we can easily prevent it.

### Terraform Version

```
Terraform v0.11.3
+ provider.aws v1.13.0
```

### Affected Resource(s)

- aws_emr_cluster

### Terraform Configuration Files

```hcl
# Omitted
```

### Panic Output

```
panic: runtime error: invalid memory address or nil pointer dereference
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1fcb0c4]
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: 
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: goroutine 152 [running]:
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsEMRClusterStateRefreshFunc.func1(0xc420041f20, 0x1, 0x0, 0x0, 0x10, 0xc4202d6ee0)
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_emr_cluster.go:1391 +0x3b4
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/resource.(*StateChangeConf).WaitForState.func1(0xc4209d7140, 0xc4201e78f0, 0xc4204dd340, 0xc4207765a0, 0xc42060a8b8, 0xc42060a8b0)
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/resource/state.go:103 +0x4e0
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: created by github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/resource.(*StateChangeConf).WaitForState
2018-04-09T18:28:19.068Z [DEBUG] plugin.terraform-provider-aws_v1.13.0_x4: 	/opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/resource/state.go:80 +0x1d0
```

### Expected Behavior

After failed cluster creation, error message output.

### Actual Behavior

Panic! 💥 

### Steps to Reproduce

1. `terraform apply`

--- a/aws/resource_aws_emr_cluster.go
+++ b/aws/resource_aws_emr_cluster.go
@@ -1390,7 +1390,7 @@ func resourceAwsEMRClusterStateRefreshFunc(d *schema.ResourceData, meta interfac
 			if reason == nil {
 				return resp.Cluster, state, fmt.Errorf("%s: reason code and message not provided", state)
 			}
-			return resp.Cluster, state, fmt.Errorf("%s: %s", aws.StringValue(reason.Code), aws.StringValue(reason.Message))
+			return resp.Cluster, state, fmt.Errorf("%s: %s: %s", state, aws.StringValue(reason.Code), aws.StringValue(reason.Message))
 		}
 
 		return resp.Cluster, state, nil

--- a/aws/resource_aws_emr_cluster.go
+++ b/aws/resource_aws_emr_cluster.go
@@ -1374,23 +1374,25 @@ func resourceAwsEMRClusterStateRefreshFunc(d *schema.ResourceData, meta interfac
 			return nil, "", err
 		}
 
-		emrc := resp.Cluster
-
-		if emrc == nil {
+		if resp.Cluster == nil {
 			return 42, "destroyed", nil
 		}
 
-		if resp.Cluster.Status != nil {
-			log.Printf("[DEBUG] EMR Cluster status (%s): %s", d.Id(), *resp.Cluster.Status)
+		if resp.Cluster.Status == nil {
+			return resp.Cluster, "", fmt.Errorf("cluster status not provided")
 		}
 
-		status := emrc.Status
-		if *status.State == "TERMINATING" || *status.State == "TERMINATED_WITH_ERRORS" {
-			reason := *status.StateChangeReason
-			return emrc, *status.State, fmt.Errorf("%s: %s",
-				*reason.Code, *reason.Message)
+		state := aws.StringValue(resp.Cluster.Status.State)
+		log.Printf("[DEBUG] EMR Cluster status (%s): %s", d.Id(), state)
+
+		if state == emr.ClusterStateTerminating || state == emr.ClusterStateTerminatedWithErrors {
+			reason := resp.Cluster.Status.StateChangeReason
+			if reason == nil {
+				return resp.Cluster, state, fmt.Errorf("%s: reason code and message not provided", state)
+			}
+			return resp.Cluster, state, fmt.Errorf("%s: %s", aws.StringValue(reason.Code), aws.StringValue(reason.Message))
 		}
 
-		return emrc, *status.State, nil
+		return resp.Cluster, state, nil
 	}
 }

